src/block.js

```diff
@@ -18,104 +18,92 @@ class Block {
     }).then(onFulfilled, onError)
   }
 
-  async getBlock(id, filterOptions) {
-    if (is.undefined(id)) return Promise.reject(new Error(NO_BLOCK_ID))
-    const response = await get(this.web3data, {
+  getBlock(id, filterOptions) {
+    throwIf(is.undefined(id), NO_BLOCK_ID)
+    return get(this.web3data, {
       pathParam: id,
       endpoint: ENDPOINT,
       filterOptions
-    })
-    return new Promise((resolve, reject) => {
-      if (!response || response.status !== 200 || !response.payload) {
-        reject(new Error(`error with request`))
-      } else {
-        resolve(response.payload)
-      }
-    })
+    }).then(onFulfilled, onError)
   }
 
-  async getBlockNumber() {
-    const block = await this.web3data.block.getBlock('latest')
-    throwIf(block | !block.number, 'Failed to retrieve block number.')
-    return parseInt(block.number, 10)
+  getBlockNumber() {
+    return this.web3data.block.getBlock('latest').then(block => {
+      throwIf(block | !block.number, 'Failed to retrieve block number.')
+      return parseInt(block.number, 10)
+    })
   }
 
-  async getBlockTransactionCount(id) {
-    const block = await this.web3data.block.getBlock(id)
-    throwIf(
-      !block || (!block.predictions && !block.numTransactions),
-      'Failed to retrieve block transaction count.'
-    )
-    return block.predictions ? null : parseInt(block.numTransactions, 10)
+  getBlockTransactionCount(id) {
+    return this.web3data.block.getBlock(id).then(block => {
+      throwIf(
+        !block || (!block.predictions && !block.numTransactions),
+        'Failed to retrieve block transaction count.'
+      )
+      // If 'predictions' field exists then it's a future block thus has no txns
+      return block.predictions ? null : parseInt(block.numTransactions, 10)
+    })
   }
 
-  async getTransactions(id, filterOptions) {
-    const response = await get(this.web3data, {
+  getTransactions(id, filterOptions) {
+    throwIf(is.undefined(id), NO_BLOCK_ID)
+    return get(this.web3data, {
       pathParam: id,
       endpoint: ENDPOINT,
       subendpoint: 'transactions',
       filterOptions
-    })
-    throwIf(
-      !response ||
-        response.status !== 200 ||
-        !response.payload ||
-        !response.payload.records,
-      'Failed to retrieve transactions.'
-    )
-    return response.payload.records
+    }).then(onFulfilled, onError)
```

Comment:
@taylorjdawson I see why you did this, however we need to discuss. I don't like forcing developers to use our non-standardized payload names. The use cases are not setup well. 
As a developer I need to:
`const transactions = await w3d.getTransactions(123456789)`
This is far more useful, and I can apply the data directly in my application.

Reply:
awesome update, resolved